c/secp256k1_blake160_sighash_all.c

```diff
@@ -131,52 +132,28 @@ int main(int argc, char* argv[])
       return ERROR_ENCODING;
     }
     args = ns(Witness_data(witness_table));
-    if (ns(Bytes_vec_len(args)) < 2) {
+    if (ns(Bytes_vec_len(args)) < 1) {
       return ERROR_WRONG_NUMBER_OF_ARGUMENTS;
     }
 
-    /* Check pubkey hash */
-    len = TEMP_SIZE;
-    ret = extract_bytes(ns(Bytes_vec_at(args, 0)), temp, &len);
-    if (ret != CKB_SUCCESS) {
-      return ERROR_ENCODING;
-    }
-    blake2b_state blake2b_ctx;
-    blake2b_init(&blake2b_ctx, BLAKE2B_BLOCK_SIZE);
-    blake2b_update(&blake2b_ctx, temp, len);
-    blake2b_final(&blake2b_ctx, temp, BLAKE2B_BLOCK_SIZE);
-
-    if (memcmp(argv[1], temp, BLAKE160_SIZE) != 0) {
-      return ERROR_PUBKEY_BLAKE160_HASH;
-    }
-
-    /* load pubkey */
-    len = TEMP_SIZE;
-    ret = extract_bytes(ns(Bytes_vec_at(args, 0)), temp, &len);
-    if (ret != CKB_SUCCESS) {
-      return ERROR_ENCODING;
-    }
-
-    secp256k1_pubkey pubkey;
-    if (secp256k1_ec_pubkey_parse(&context, &pubkey, temp, len) == 0) {
-      return ERROR_SECP_PARSE_PUBKEY;
-    }
-
     /* Load signature */
     len = TEMP_SIZE;
-    ret = extract_bytes(ns(Bytes_vec_at(args, 1)), temp, &len);
+    ret = extract_bytes(ns(Bytes_vec_at(args, 0)), temp, &len);
     if (ret != CKB_SUCCESS) {
       return ERROR_ENCODING;
     }
 
-    secp256k1_ecdsa_signature signature;
-    if (secp256k1_ecdsa_signature_parse_der(&context, &signature, temp, len) == 0) {
+    /* The 65th byte is recid according to contract spec.*/
+    recid = temp[RECID_INDEX];
+    /* Recover pubkey */
+    secp256k1_ecdsa_recoverable_signature signature;
+    if (secp256k1_ecdsa_recoverable_signature_parse_compact(&context, &signature, temp, recid) == 0) {
```

Comment:
Is there a way we can keep DER format? There were discussions on this, and we decided we want to move forward with DER format, not compact format.

Reply:
seck256k1 provides the DER serialize methods, however, it will slightly increase the signature size.
I am curious about the reason that we decide to use  DER format. @janx

Reply:
we had a discussion and the conclusion is to go with compact format.